blog-tech/2022-05-07-connect-private-driver.md

```diff
@@ -0,0 +1,537 @@
+# Integrate a Private Driver
+
+In this article, we will first write a `GPIO` driver to control `LED` using `Python`, and then connect it to ***Shifu*** for interaction and management.
+
+## Create the driver
+
+### Objective
+
+- Complete simple LED circuit connection
+- Basic Raspberry Pi/SSH configuration
+- Basic Python syntax and knowledge of GPIO libraries
+
+### Devices
+
+- Raspberry Pi `Raspberry Pi 3B+` running 64-bit `Raspberry Pi OS`
+- 1 breadboard
+- 3 LED bulbs (red, yellow, green)
+- 1 x 330 ohm resistor
+
+### Basic knowledge
+
+- Simple Python syntax
+- Basic Linux command line operations (create files, install apps, SSH, run programs)
+
+### First step Circuit design
+
+First let's design the circuit. We need to design a circuit that will allow the `GPIO` output of the Raspberry Pi to control the on/off of a single LED bulb. In this article, we have used the most straightforward approach, i.e., using `GPIO` directly to power the LED bulbs.
+
+The circuit diagram is as follows.
+
+<img src="/blog-220507/1.png" width="50%" />
+
+The `GPIO` pins 22, 23, and 19 control the red, green, and yellow LEDs respectively. 330 ohms resistors are connected in series at the end to prevent the LEDs from burning out due to excessive current.
+
+### Step 2 Circuit Implementation
+
+According to the Pin Layout in the official `Raspberry Pi` documentation, we can see the exact location of the pins, here pins 15, 16, 35 and 39 are used.
+
+<img src="/blog-220507/2.png" width="100%" />
+
+Connect these four pins to the female port of the cable, and then connect the remaining circuits on the breadboard to.
+
+<img src="/blog-220507/3.png" width="100%" />
+
+The red, green, yellow, and gray cables in the diagram correspond to `GPIO22` `GPIO23` `GPIO19` and ground (ground), respectively.
+
+<img src="/blog-220507/4.png" width="50%" />
+
+<img src="/blog-220507/5.png" width="50%" />
+
+This concludes the circuit design and connectivity section.
+
+### Step 3 Raspberry Pi Preparation
+
+First install an operating system in the Raspberry Pi, the one used in this article is `Raspberry Pi OS (64-bit)`. [download link](https://www.raspberrypi.com/software/operating-systems/#raspberry-pi-os-64-bit)
+
+Insert the SD card into the reader, connect it to the USB port of your computer, and then swipe the downloaded zip file into the SD card via `balenaEtcher`. [balenaEtcher link](https://www.balena.io/etcher/)
+
+Insert the SD card into the Raspberry Pi, plug in the power, and the monitor is ready to start configuration.
```

Comment:
...plug in the power, and the monitor is ready to start configuration. => plug in the power and monitor cable, and we can start the configuration